pddl/parser/domain.py

```diff
@@ -157,6 +157,24 @@ def gd(self, args):
             ):
                 raise PDDLMissingRequirementError(Requirements.DIS_PRECONDITION)
             return Imply(args[2], args[3])
+        elif args[1] == Symbols.FORALL.value:
+            if not bool(
+                {Requirements.UNIVERSAL_PRECONDITION, Requirements.QUANTIFIED_PRECONDITION, Requirements.ADL}
+                & self._extended_requirements
+            ):
+                raise PDDLMissingRequirementError(Requirements.UNIVERSAL_PRECONDITION)
+            variables = [Variable(name, tags) for name, tags in args[3].items()]
+            condition = args[5]
```

Comment:
These two lines are also repeated in 175-176. Although it is totally fine to have both of them, an idea could be to wrap the code in a function call to avoid repetition.